File: Core/Workspace/Typescript.legacy/Site/wwwroot/Core/Workspace/Protocol/Push/PushRequest.ts
namespace Allors.Protocol {
    export class PushRequest {
        newObjects: PushRequestNewObject[];
        objects: PushRequestObject[];

        constructor(fields?: Partial<PushRequest>) {
          Object.assign(this, fields);
        }
    }

    export class PushRequestNewObject {
      ni: string;
      t: string;
      roles: PushRequestRole[];
    }

    export class PushRequestObject {
      i: string;
      v: string;
      roles: PushRequestRole[];
    }

    export class PushRequestRole {
        t: string;
        s: any;
        a: string[];
        r: string[];
    }
}

File: Platform/Protocol/Allors.Protocol/Api/Push/PushRequest.cs
// <copyright file="PushRequest.cs" company="Allors bv">
// Copyright (c) Allors bv. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Protocol.Remote.Push
{
    public class PushRequest
    {
        /// <summary>
        /// New Objects
        /// </summary>
        public PushRequestNewObject[] newObjects { get; set; }

        /// <summary>
        /// Objects
        /// </summary>
        public PushRequestObject[] objects { get; set; }
    }
}
